planetary-social/nos
# Refactor Bech32 and TLV usage

## Issues covered
#1294 

## Description
This removes direct usage of Bech32 and TLV in favor of NostrIdentifier. It removes TLV entirely, as it's been superseded by TLVElement. Now, the only places we use TLVElement are in NostrIdentifier. And the only place we use Bech32.decode is in NostrIdentifier.

There's still a Bech32.encode method that's used in three places. We could replace those with a not-yet-written encode method on NostrIdentifier. I think that would be fairly simple.

## How to test
1. Examine the unit tests. Run them, too.
2. Make sure the app still works, especially around keys and `npub`s and `note`s.

## To Do

- [x] Fix issue with `nostr:naddr` links jumping out to another app.
- [ ] Consider moving Bech32.encode into NostrIdentifier, and do it if it makes sense.
- [x] Rename NostrConstants.swift, or delete it and move the constants to NostrIdentifier, or something. Those constants are 100% related to NostrIdentifier and nothing else.
